agent/consul/leader_test.go

```diff
@@ -1272,6 +1273,99 @@ func TestLeader_ACLUpgrade(t *testing.T) {
 	})
 }
 
+func TestLeader_ACLUpgrade_IsStickyEvenIfSerfTagsRegress(t *testing.T) {
+	if testing.Short() {
+		t.Skip("too slow for testing.Short")
+	}
+
+	// t.Parallel()
+
+	// We test this by having two datacenters with one server each. They
```

Comment:
This scenario is actually the thing I was running into while trying to fix #9342. Without this fix, the patches on the other PR to prevent dialing the primary's gateway when the local is known to have a gateway were blocked waiting for the mesh gateway envoy to start in the secondary. That in turn was blocked on the "wtf? i can't handle non-legacy tokens yet until i talk to the primary" circular dependency.